functions/go/apply-setter/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+//nolint
+func main() {
+	resourceList := &framework.ResourceList{}
+	resourceList.FunctionConfig = map[string]interface{}{}
+
+	cmd := framework.Command(resourceList, func() error {
+		s, err := getSetters(resourceList.FunctionConfig)
+		if err != nil {
+			return fmt.Errorf("failed to get data.setters field from function config: %w", err)
+		}
+		_, err = s.Filter(resourceList.Items)
+		if err != nil {
+			return fmt.Errorf("failed to set setters: %w", err)
+		}
+		return nil
+	})
+
+	cmd.Long = usage()
+	if err := cmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func usage() string {
+	return `Set values of resources with setter values.
+
+Configured using a ConfigMap with the following keys:
+
+name: Setter name.
+value: Setter value.
+
+These keys are in a list in path 'data.setters'.
+
+Example:
+
+To set a setter 'color: orange' to all resources:
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  setters:
+  - name: color
+	  value: orange
+
+  To set 2 setters 'color: orange' and 'fruit: apple' to all resources:
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+setters:
+  - name: color
+    value: orange
+  - name: fruit
+    value: apple
+`
+}
+
+func getSetters(fc interface{}) (Set, error) {
+	var fcd Set
+	f, ok := fc.(map[string]interface{})
+	if !ok {
+		return fcd, fmt.Errorf("function config %#v is not valid", fc)
+	}
+	rn, err := kyaml.FromMap(f)
+	if err != nil {
+		return fcd, fmt.Errorf("failed to parse from function config: %w", err)
```

Comment:
What does `%w` do?

Reply:
https://blog.golang.org/go1.13-errors

Wrapping an error with %w makes it available to errors.Is and errors.As: